### ISSUE TITLE
Several Unknown Entities (Mostly Odometer-related) on Polestar 3 (is this expected)?

### ISSUE THREAD

**Issue by BJReplay** [pypolestar/polestar_api#255]
Hi team, I've finally got my 3, and having watched it spring to life a couple of days ago, to having picked it up yesterday, I have now watched the charge level degrade by 1-2% every 12 hours 😢, and a few faults, so I'll be back to Polestar on Monday.

Like a new kid on the block, I want to see what I can see - some stuff is unknown.

This is what shows up:

![image](https://github.com/user-attachments/assets/19a94844-85cc-48a6-b395-9bdc13bfd122)

Below is a partially redacted log: 

(I'm wondering if VIN, access and refresh tokens should be redacted when logging):

This is running the combined PRs for #249 and #254 (merged into my Dev Container), but I see the same missing sensors running 1.10.0

Sensors are missing after an integration reload as well as a brand-new set up of the integration.

I've redacted a few things and deleted a second auth log.

Since I'm part of the team and keen to help, is there anything else that I should do to start looking at this?

I can feel my way around postman and Bearer auth, but have never done anything with GraphQL before, but happy to help in any way I can.

``` log
2024-12-07 11:42:05.729 DEBUG (MainThread) [custom_components.polestar_api] async_unload_entry: <ConfigEntry entry_id=01JEF8QF6WZEHQ7X6YCXJM000H version=1 domain=polestar_api title=Polestar EV for polestar@id.redacted.com state=ConfigEntryState.LOADED unique_id=None>
2024-12-07 11:42:05.837 DEBUG (MainThread) [custom_components.polestar_api] async_setup_entry: <ConfigEntry entry_id=01JEF8QF6WZEHQ7X6YCXJM000H version=1 domain=polestar_api title=Polestar EV for polestar@id.redacted.com state=ConfigEntryState.SETUP_IN_PROGRESS unique_id=None>
2024-12-07 11:42:05.837 DEBUG (MainThread) [custom_components.polestar_api.polestar] Configure Polestar API client for car with VIN VINREDACTED
2024-12-07 11:42:07.833 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.auth.01JEF8QF6WZEHQ7X6YCXJM000H] Auth Token Result: {'getAuthToken': {'id_token': None, 'access_token': 'access token redacted', 'refresh_token': 'refresh token redacted', 'expires_in': 299}}
2024-12-07 11:42:07.834 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL URL: https://pc-api.polestar.com/eu-north-1/mystar-v2/
2024-12-07 11:42:09.092 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL Result: {'getConsumerCarsV2': [{'vin': 'VINREDACTED', 'internalVehicleIdentifier': 'd3b461ca-2cde-462c-8d8f-387ced7b6138', 'registrationNo': 'REGO-RED', 'registrationDate': None, 'factoryCompleteDate': '2024-06-20', 'content': {'model': {'name': 'Polestar 3'}, 'images': {'studio': {'url': 'https://cas.polestar.com/image/dynamic/MY24_2207/359/summary-transparent-v2/EA/2/73600/RCJ000/R102/LR02/EV02/K503/JB02/SW01/_/ET01/default.png?market=au'}}, 'specification': {'battery': '400V lithium-ion battery, 111 kWh capacity, 17 modules', 'torque': '840 Nm / 620 lbf-ft'}}, 'software': {'version': None, 'versionTimestamp': None}}]}
2024-12-07 11:42:09.093 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] API setup for VIN VINREDACTED
2024-12-07 11:42:09.093 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/model/name
2024-12-07 11:42:09.094 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Starting update for VIN VINREDACTED
2024-12-07 11:42:09.094 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL URL: https://pc-api.polestar.com/eu-north-1/mystar-v2/
2024-12-07 11:42:09.783 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL Result: {'getOdometerData': None}
2024-12-07 11:42:09.783 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Received odometer data: {'data': None, 'timestamp': datetime.datetime(2024, 12, 7, 11, 42, 9, 783829)}
2024-12-07 11:42:09.784 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL URL: https://pc-api.polestar.com/eu-north-1/mystar-v2/
2024-12-07 11:42:11.040 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL Result: {'getBatteryData': {'averageEnergyConsumptionKwhPer100Km': None, 'batteryChargeLevelPercentage': 71, 'chargerConnectionStatus': 'CHARGER_CONNECTION_STATUS_CONNECTED', 'chargingCurrentAmps': 9, 'chargingPowerWatts': 2050, 'chargingStatus': 'CHARGING_STATUS_CHARGING', 'estimatedChargingTimeMinutesToTargetDistance': 0, 'estimatedChargingTimeToFullMinutes': 540, 'estimatedDistanceToEmptyKm': 370, 'estimatedDistanceToEmptyMiles': 230, 'eventUpdatedTimestamp': {'iso': '2024-12-07T00:30:36.000Z', 'unix': '1733531436'}}}
2024-12-07 11:42:11.041 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Received battery data: {'data': {'averageEnergyConsumptionKwhPer100Km': None, 'batteryChargeLevelPercentage': 71, 'chargerConnectionStatus': 'CHARGER_CONNECTION_STATUS_CONNECTED', 'chargingCurrentAmps': 9, 'chargingPowerWatts': 2050, 'chargingStatus': 'CHARGING_STATUS_CHARGING', 'estimatedChargingTimeMinutesToTargetDistance': 0, 'estimatedChargingTimeToFullMinutes': 540, 'estimatedDistanceToEmptyKm': 370, 'estimatedDistanceToEmptyMiles': 230, 'eventUpdatedTimestamp': {'iso': '2024-12-07T00:30:36.000Z', 'unix': '1733531436'}}, 'timestamp': datetime.datetime(2024, 12, 7, 11, 42, 11, 41019)}
2024-12-07 11:42:11.041 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Update took 1.95 seconds
2024-12-07 11:42:11.041 DEBUG (MainThread) [custom_components.polestar_api] Added car with VIN VINREDACTED for 01JEF8QF6WZEHQ7X6YCXJM000H
2024-12-07 11:42:11.051 INFO (MainThread) [homeassistant.components.image] Setting up polestar_api.image
2024-12-07 11:42:11.071 INFO (MainThread) [homeassistant.components.sensor] Setting up polestar_api.sensor
2024-12-07 11:42:11.077 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:42:11.078 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:42:11.078 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData odometerMeters
2024-12-07 11:42:11.078 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData odometerMeters
2024-12-07 11:42:11.078 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData odometerMeters
2024-12-07 11:42:11.078 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData odometerMeters
2024-12-07 11:42:11.079 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:42:11.079 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:42:11.079 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:42:11.079 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:42:11.079 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:42:11.079 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:42:11.080 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:42:11.080 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:42:11.080 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:42:11.080 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:42:11.080 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:42:11.080 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:42:11.080 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData batteryChargeLevelPercentage
2024-12-07 11:42:11.081 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData batteryChargeLevelPercentage
2024-12-07 11:42:11.081 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:42:11.081 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:42:11.081 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingStatus
2024-12-07 11:42:11.081 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingStatus
2024-12-07 11:42:11.081 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingPowerWatts
2024-12-07 11:42:11.081 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingPowerWatts
2024-12-07 11:42:11.082 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingCurrentAmps
2024-12-07 11:42:11.082 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingCurrentAmps
2024-12-07 11:42:11.082 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargerConnectionStatus
2024-12-07 11:42:11.082 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargerConnectionStatus
2024-12-07 11:42:11.082 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData averageEnergyConsumptionKwhPer100Km
2024-12-07 11:42:11.083 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData averageEnergyConsumptionKwhPer100Km
2024-12-07 11:42:11.083 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeMinutesToTargetDistance
2024-12-07 11:42:11.084 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeMinutesToTargetDistance
2024-12-07 11:42:11.084 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 vin
2024-12-07 11:42:11.085 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 vin
2024-12-07 11:42:11.085 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/version
2024-12-07 11:42:11.085 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/version
2024-12-07 11:42:11.085 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/versionTimestamp
2024-12-07 11:42:11.085 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/versionTimestamp
2024-12-07 11:42:11.086 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationDate
2024-12-07 11:42:11.086 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationDate
2024-12-07 11:42:11.086 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationNo
2024-12-07 11:42:11.086 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationNo
2024-12-07 11:42:11.086 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 factoryCompleteDate
2024-12-07 11:42:11.087 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 factoryCompleteDate
2024-12-07 11:42:11.087 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 internalVehicleIdentifier
2024-12-07 11:42:11.087 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 internalVehicleIdentifier
2024-12-07 11:42:11.087 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:42:11.087 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:42:11.088 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/model/name
2024-12-07 11:42:11.088 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/model/name
2024-12-07 11:42:11.089 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:42:11.092 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:42:11.092 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:42:11.093 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:42:11.093 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData eventUpdatedTimestamp/iso
2024-12-07 11:42:11.093 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData eventUpdatedTimestamp/iso
2024-12-07 11:42:11.094 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:42:11.094 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:42:11.094 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/torque
2024-12-07 11:42:11.095 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/torque
2024-12-07 11:42:11.095 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/battery
2024-12-07 11:42:11.095 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/battery
2024-12-07 11:42:11.224 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 factoryCompleteDate
2024-12-07 11:42:11.231 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationDate
2024-12-07 11:42:11.244 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData batteryChargeLevelPercentage
2024-12-07 11:42:11.245 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:42:11.255 INFO (MainThread) [homeassistant.components.binary_sensor] Setting up polestar_api.binary_sensor
2024-12-07 11:42:11.273 WARNING (MainThread) [asyncio] Executing <Task finished name='Task-1432' coro=<RequestHandler._handle_request() done, defined at /home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/web_protocol.py:467> result=(<Response OK eof>, False) created at /home/vscode/.local/ha-venv/lib/python3.12/site-packages/aiohttp/web_protocol.py:556> took 0.234 seconds
2024-12-07 11:43:11.254 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Starting update for VIN VINREDACTED
2024-12-07 11:43:11.254 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL URL: https://pc-api.polestar.com/eu-north-1/mystar-v2/
2024-12-07 11:43:11.258 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData odometerMeters
2024-12-07 11:43:11.258 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData odometerMeters
2024-12-07 11:43:11.259 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:43:11.259 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:43:11.260 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:43:11.260 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:43:11.261 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:43:11.261 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:43:11.262 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData batteryChargeLevelPercentage
2024-12-07 11:43:11.263 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:43:11.264 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingStatus
2024-12-07 11:43:11.265 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingPowerWatts
2024-12-07 11:43:11.266 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingCurrentAmps
2024-12-07 11:43:11.268 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargerConnectionStatus
2024-12-07 11:43:11.269 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData averageEnergyConsumptionKwhPer100Km
2024-12-07 11:43:11.270 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeMinutesToTargetDistance
2024-12-07 11:43:11.271 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 vin
2024-12-07 11:43:11.271 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 factoryCompleteDate
2024-12-07 11:43:11.273 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/version
2024-12-07 11:43:11.274 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/versionTimestamp
2024-12-07 11:43:11.275 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationNo
2024-12-07 11:43:11.275 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationDate
2024-12-07 11:43:11.277 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:43:11.281 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/model/name
2024-12-07 11:43:11.282 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:43:11.283 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:43:11.284 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData eventUpdatedTimestamp/iso
2024-12-07 11:43:11.287 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:43:11.287 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData batteryChargeLevelPercentage
2024-12-07 11:43:11.287 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:43:11.291 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/torque
2024-12-07 11:43:11.292 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/battery
2024-12-07 11:43:11.477 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL Result: {'getOdometerData': None}
2024-12-07 11:43:11.477 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Received odometer data: {'data': None, 'timestamp': datetime.datetime(2024, 12, 7, 11, 43, 11, 477589)}
2024-12-07 11:43:11.477 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL URL: https://pc-api.polestar.com/eu-north-1/mystar-v2/
2024-12-07 11:43:11.553 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL Result: {'getBatteryData': {'averageEnergyConsumptionKwhPer100Km': None, 'batteryChargeLevelPercentage': 71, 'chargerConnectionStatus': 'CHARGER_CONNECTION_STATUS_CONNECTED', 'chargingCurrentAmps': 9, 'chargingPowerWatts': 2050, 'chargingStatus': 'CHARGING_STATUS_CHARGING', 'estimatedChargingTimeMinutesToTargetDistance': 0, 'estimatedChargingTimeToFullMinutes': 540, 'estimatedDistanceToEmptyKm': 370, 'estimatedDistanceToEmptyMiles': 230, 'eventUpdatedTimestamp': {'iso': '2024-12-07T00:30:36.000Z', 'unix': '1733531436'}}}
2024-12-07 11:43:11.553 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Received battery data: {'data': {'averageEnergyConsumptionKwhPer100Km': None, 'batteryChargeLevelPercentage': 71, 'chargerConnectionStatus': 'CHARGER_CONNECTION_STATUS_CONNECTED', 'chargingCurrentAmps': 9, 'chargingPowerWatts': 2050, 'chargingStatus': 'CHARGING_STATUS_CHARGING', 'estimatedChargingTimeMinutesToTargetDistance': 0, 'estimatedChargingTimeToFullMinutes': 540, 'estimatedDistanceToEmptyKm': 370, 'estimatedDistanceToEmptyMiles': 230, 'eventUpdatedTimestamp': {'iso': '2024-12-07T00:30:36.000Z', 'unix': '1733531436'}}, 'timestamp': datetime.datetime(2024, 12, 7, 11, 43, 11, 553531)}
2024-12-07 11:43:11.553 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Update took 0.30 seconds
2024-12-07 11:43:11.553 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:44:11.254 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Starting update for VIN VINREDACTED
2024-12-07 11:44:11.255 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL URL: https://pc-api.polestar.com/eu-north-1/mystar-v2/
2024-12-07 11:44:11.258 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData odometerMeters
2024-12-07 11:44:11.258 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData odometerMeters
2024-12-07 11:44:11.259 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:44:11.259 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:44:11.259 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:44:11.259 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:44:11.259 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:44:11.259 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:44:11.260 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData batteryChargeLevelPercentage
2024-12-07 11:44:11.261 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:44:11.263 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingStatus
2024-12-07 11:44:11.265 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingPowerWatts
2024-12-07 11:44:11.266 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingCurrentAmps
2024-12-07 11:44:11.267 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargerConnectionStatus
2024-12-07 11:44:11.269 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData averageEnergyConsumptionKwhPer100Km
2024-12-07 11:44:11.270 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeMinutesToTargetDistance
2024-12-07 11:44:11.271 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 vin
2024-12-07 11:44:11.271 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 factoryCompleteDate
2024-12-07 11:44:11.275 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/version
2024-12-07 11:44:11.276 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/versionTimestamp
2024-12-07 11:44:11.278 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationNo
2024-12-07 11:44:11.278 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationDate
2024-12-07 11:44:11.281 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:44:11.284 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/model/name
2024-12-07 11:44:11.285 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:44:11.285 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:44:11.287 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData eventUpdatedTimestamp/iso
2024-12-07 11:44:11.289 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:44:11.289 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData batteryChargeLevelPercentage
2024-12-07 11:44:11.290 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:44:11.292 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/torque
2024-12-07 11:44:11.295 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/battery
2024-12-07 11:44:11.440 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL Result: {'getOdometerData': None}
2024-12-07 11:44:11.441 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Received odometer data: {'data': None, 'timestamp': datetime.datetime(2024, 12, 7, 11, 44, 11, 441048)}
2024-12-07 11:44:11.441 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL URL: https://pc-api.polestar.com/eu-north-1/mystar-v2/
2024-12-07 11:44:11.563 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL Result: {'getBatteryData': {'averageEnergyConsumptionKwhPer100Km': None, 'batteryChargeLevelPercentage': 71, 'chargerConnectionStatus': 'CHARGER_CONNECTION_STATUS_CONNECTED', 'chargingCurrentAmps': 9, 'chargingPowerWatts': 2050, 'chargingStatus': 'CHARGING_STATUS_CHARGING', 'estimatedChargingTimeMinutesToTargetDistance': 0, 'estimatedChargingTimeToFullMinutes': 540, 'estimatedDistanceToEmptyKm': 370, 'estimatedDistanceToEmptyMiles': 230, 'eventUpdatedTimestamp': {'iso': '2024-12-07T00:30:36.000Z', 'unix': '1733531436'}}}
2024-12-07 11:44:11.563 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Received battery data: {'data': {'averageEnergyConsumptionKwhPer100Km': None, 'batteryChargeLevelPercentage': 71, 'chargerConnectionStatus': 'CHARGER_CONNECTION_STATUS_CONNECTED', 'chargingCurrentAmps': 9, 'chargingPowerWatts': 2050, 'chargingStatus': 'CHARGING_STATUS_CHARGING', 'estimatedChargingTimeMinutesToTargetDistance': 0, 'estimatedChargingTimeToFullMinutes': 540, 'estimatedDistanceToEmptyKm': 370, 'estimatedDistanceToEmptyMiles': 230, 'eventUpdatedTimestamp': {'iso': '2024-12-07T00:30:36.000Z', 'unix': '1733531436'}}, 'timestamp': datetime.datetime(2024, 12, 7, 11, 44, 11, 563534)}
2024-12-07 11:44:11.563 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Update took 0.31 seconds
2024-12-07 11:44:11.563 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:45:11.256 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Starting update for VIN VINREDACTED
2024-12-07 11:45:11.257 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.auth.01JEF8QF6WZEHQ7X6YCXJM000H] Token expires in 115 seconds, time to refresh
2024-12-07 11:45:11.262 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData odometerMeters
2024-12-07 11:45:11.262 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData odometerMeters
2024-12-07 11:45:11.263 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:45:11.263 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData averageSpeedKmPerHour
2024-12-07 11:45:11.265 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:45:11.265 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterAutomaticKm
2024-12-07 11:45:11.267 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:45:11.267 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData tripMeterManualKm
2024-12-07 11:45:11.269 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData batteryChargeLevelPercentage
2024-12-07 11:45:11.270 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:45:11.272 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingStatus
2024-12-07 11:45:11.274 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingPowerWatts
2024-12-07 11:45:11.275 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargingCurrentAmps
2024-12-07 11:45:11.277 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData chargerConnectionStatus
2024-12-07 11:45:11.278 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData averageEnergyConsumptionKwhPer100Km
2024-12-07 11:45:11.280 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeMinutesToTargetDistance
2024-12-07 11:45:11.282 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 vin
2024-12-07 11:45:11.282 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 factoryCompleteDate
2024-12-07 11:45:11.285 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/version
2024-12-07 11:45:11.287 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 software/versionTimestamp
2024-12-07 11:45:11.289 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationNo
2024-12-07 11:45:11.289 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 registrationDate
2024-12-07 11:45:11.291 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedChargingTimeToFullMinutes
2024-12-07 11:45:11.294 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/model/name
2024-12-07 11:45:11.295 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:45:11.295 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data returning None for VINREDACTED getOdometerData eventUpdatedTimestamp/iso
2024-12-07 11:45:11.297 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData eventUpdatedTimestamp/iso
2024-12-07 11:45:11.299 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:45:11.300 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData batteryChargeLevelPercentage
2024-12-07 11:45:11.300 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
2024-12-07 11:45:11.303 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/torque
2024-12-07 11:45:11.305 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getConsumerCarsV2 content/specification/battery
2024-12-07 11:45:12.840 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL URL: https://pc-api.polestar.com/eu-north-1/mystar-v2/
2024-12-07 11:45:14.276 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL Result: {'getOdometerData': None}
2024-12-07 11:45:14.276 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Received odometer data: {'data': None, 'timestamp': datetime.datetime(2024, 12, 7, 11, 45, 14, 276696)}
2024-12-07 11:45:14.276 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL URL: https://pc-api.polestar.com/eu-north-1/mystar-v2/
2024-12-07 11:45:15.541 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] GraphQL Result: {'getBatteryData': {'averageEnergyConsumptionKwhPer100Km': None, 'batteryChargeLevelPercentage': 71, 'chargerConnectionStatus': 'CHARGER_CONNECTION_STATUS_CONNECTED', 'chargingCurrentAmps': 9, 'chargingPowerWatts': 2050, 'chargingStatus': 'CHARGING_STATUS_CHARGING', 'estimatedChargingTimeMinutesToTargetDistance': 0, 'estimatedChargingTimeToFullMinutes': 540, 'estimatedDistanceToEmptyKm': 370, 'estimatedDistanceToEmptyMiles': 230, 'eventUpdatedTimestamp': {'iso': '2024-12-07T00:30:36.000Z', 'unix': '1733531436'}}}
2024-12-07 11:45:15.541 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Received battery data: {'data': {'averageEnergyConsumptionKwhPer100Km': None, 'batteryChargeLevelPercentage': 71, 'chargerConnectionStatus': 'CHARGER_CONNECTION_STATUS_CONNECTED', 'chargingCurrentAmps': 9, 'chargingPowerWatts': 2050, 'chargingStatus': 'CHARGING_STATUS_CHARGING', 'estimatedChargingTimeMinutesToTargetDistance': 0, 'estimatedChargingTimeToFullMinutes': 540, 'estimatedDistanceToEmptyKm': 370, 'estimatedDistanceToEmptyMiles': 230, 'eventUpdatedTimestamp': {'iso': '2024-12-07T00:30:36.000Z', 'unix': '1733531436'}}, 'timestamp': datetime.datetime(2024, 12, 7, 11, 45, 15, 541765)}
2024-12-07 11:45:15.541 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] Update took 4.28 seconds
2024-12-07 11:45:15.541 DEBUG (MainThread) [custom_components.polestar_api.pypolestar.polestar.01JEF8QF6WZEHQ7X6YCXJM000H] get_latest_data VINREDACTED getBatteryData estimatedDistanceToEmptyKm
```

**Comment by jschlyter** [comment]
As seen in the debug log, the API returns `None` for the odometers. I've included some additional diagnostics data in #257 - can you try that and she the results of the diagnostics? (PM via mail works)

**Comment by jschlyter** [comment]
Looking at the diagnostics output you sent me via email I can see that the odometer data is not available at all, one can also see this in the logs above from `GraphQL Result: {'getOdometerData': None}`. Battery seems fine. Not sure why this would happen, I guess you can see the odometer in the app?

**Comment by BJReplay** [comment]
Yes. Running my phone through Adguard Home, the second screenshot shows what DNS calls are captured as I open the app and display the odometers. 

I wonder if that suggests that a second endpoint is needed for odometer data. I have a rooted phone that I might be able to use to get further information on the details of calls being used, but it is not something that I am familiar with, but I will do some reading and see what I can find. 

![Screenshot_20241209_071730.jpg](https://github.com/user-attachments/assets/5498744f-6818-4b14-9dc3-7eec2d86466d)

![Screenshot_20241209_071713.jpg](https://github.com/user-attachments/assets/e0fc12f7-137e-4fbd-8c17-9afaa3094903)

**Comment by BJReplay** [comment]
Loaded the Polestar app onto a rooted phone, installed HTTP Toolkit, and found some things:

`POST https://cepmobtoken.kr.prod.c3.volvocars.com/services.vehiclestates.odometer.OdometerService/GetOdometer`

content-type `application/grpc`

Request body 

``` json
{
  "2": "VIN redacted"
}
```

Bearer - same as the `POST https://pc-api.polestar.com/eu-north-1/mesh/` so one login looks like it will work.

Returns 

``` json
{
  "2": "VIN redacted",
  "3": {
    "1": {
      "1": 1733700266,
      "2": 788576000
    },
    "2": 101000,
    "3": {
      "Type": "Buffer (4 bytes)",
      "As string": "\u0000\u0000ÊB",
      "As hex": "00 00 ca 42"
    },
    "4": {
      "Type": "Buffer (4 bytes)",
      "As string": "\u0000\u0000\u0000\u0000",
      "As hex": "00 00 00 00"
    },
    "5": 0,
    "6": 30,
    "7": {
      "Type": "Buffer (4 bytes)",
      "As string": "ffÆ@",
      "As hex": "66 66 c6 40"
    },
    "8": 28
  }
}
```

Lots more, but mainly repeating calls as noted below.

Notes:

2: is total distance travelled in meters
8: appears to be average speed since last charge in km/h
6: average speed last trip (Trip Auto) in km/h
5: could be Distance (Last Trip - TA) or Average Speed (TM)

I haven't attempted to decode 3, 4, or 7.

Mainly multiple calls to (POSTS) 
```
https://cepmobtoken.kr.prod.c3.volvocars.com/services.vehiclestates.battery.BatteryService/GetBattery
https://cepmobtoken.kr.prod.c3.volvocars.com/services.vehiclestates.odometer.OdometerService/GetOdometer
```

Screenshot of app at same time as captured calls shown.

![Screenshot_20241209_131029](https://github.com/user-attachments/assets/bf333da8-f8af-4299-8c9c-e426c618dfd0)


So, it looks like the app calls an API at volvocars.com rather than polestar.com

Given the nature of the API, happy if you want to close.

**Comment by jschlyter** [comment]
This is the Android app? Seems to run gRPC.

**Comment by BJReplay** [comment]
> This is the Android app? Seems to run gRPC (protobuf).

Yep, Android. Don't have an iPhone or iOS device to test, sorry.

I did start off trying to get an Android 13 emulator running (since, in theory, Android 14 made certificate interception difficult), but as it turns out, HTTP Toolkit found a way around on rooted devices or emulators, and by the time I'd got it going on a device, I'd given up on Android Studio.

**Comment by BJReplay** [comment]
Just to add to this - I'm going to follow up with Polestar, because I noticed this the other day:

![image](https://github.com/user-attachments/assets/aee7956d-5050-4512-a08f-601dbbfa7919)

When I click on the `?` symbol, this is displayed:

![image](https://github.com/user-attachments/assets/f377cef5-b651-4345-bc9c-ffa0b1dad344)

There's neither Polestar Connect (that I can find) in the settings menu in the car, nor Data sharing in the Polestar app.

**Comment by jschlyter** [comment]
Please try https://github.com/pypolestar/polestar_api/releases/tag/1.14.0 and report back (even though we both believe this is a problem with your Polestar ID and not a general issue).

**Comment by BJReplay** [comment]
Confirmed still not working with 1.14.0.

Health data shows up as Unknown as well.

Still waiting on Polestar 😢

**Comment by jschlyter** [comment]
I'm closing this as everything indicates it's a Polestar problem.
